src/integrations/prefect-kubernetes/prefect_kubernetes/worker.py

```diff
@@ -621,7 +646,134 @@ async def run(
 
             return KubernetesWorkerResult(identifier=pid, status_code=status_code)
 
-    async def teardown(self, *exc_info):
+    async def submit(
+        self,
+        flow: "Flow[..., R]",
+        parameters: dict[str, Any] | None = None,
+        job_variables: dict[str, Any] | None = None,
+    ) -> "PrefectFlowRunFuture[R]":
+        """
+        EXPERIMENTAL: The interface for this method is subject to change.
+
+        Submits a flow to run in a Kubernetes job.
+
+        Args:
+            flow: The flow to submit
+            parameters: The parameters to pass to the flow
+
+        Returns:
+            A flow run object
+        """
+        if self._runs_task_group is None:
+            raise RuntimeError("Worker not properly initialized")
+        flow_run = await self._runs_task_group.start(
```

Comment:
i dont see a return statement in `_submit_adhoc_run`?

Reply:
Since we start this with an `anyio` task group, we can emit a value back via the `task_status` that is passed into `_submit_adhoc_run`. This allows us to emit the flow run after it is created and return a future before submission is completed. I added a comment for this in [d18f090](https://github.com/PrefectHQ/prefect/pull/17218/commits/d18f090343cde29cf037a7b11146ff684d7f1051).